cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -535,6 +541,166 @@ public void programSpecification(HttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, specification);
   }
 
+  @PUT
+  @Path("apps/{app-name}/{program-type}/{program-name}/schedules/{schedule-name}")
+  @AuditPolicy(AuditDetail.REQUEST_BODY)
+  public void addSchedule(HttpRequest request, HttpResponder responder,
+                          @PathParam("namespace-id") String namespaceId,
+                          @PathParam("app-name") String appName,
+                          @PathParam("program-type") String programType,
+                          @PathParam("program-name") String programName,
+                          @PathParam("schedule-name") String scheduleName)
+    throws SchedulerException, BadRequestException, NotFoundException, IOException, AlreadyExistsException {
+    doAddSchedule(request, responder, namespaceId, appName, ApplicationId.DEFAULT_VERSION,
+                  programType, programName, scheduleName);
+  }
+
+  @PUT
+  @Path("apps/{app-name}/versions/{app-version}/{program-type}/{program-name}/schedules/{schedule-name}")
```

Comment:
We support scheduling of program types that are of `SchedulableProgramType` - WORKFLOW, MAPREDUCE, SPARK, CUSTOM_ACTION